components/script/dom/document.rs

```diff
@@ -1650,13 +1650,15 @@ impl Document {
                             node.get_stylesheet()
                         } else {
                             None
-                        }
+                        }.map(|stylesheet| (JS::from_rooted(&node), stylesheet))
                     })
-                    .collect();
-                *stylesheets = Some(new_stylesheets);
+                    .collect());
             };
         }
-        Ref::map(self.stylesheets.borrow(), |t| t.as_ref().unwrap())
+        self.stylesheets.borrow().clone()
+                        .unwrap().into_iter()
```

Comment:
You can use iter() and clone on the map; that will only allocate one vector rather than two in this expression.